core/notifications/proto/notifications.proto

```diff
@@ -238,17 +238,31 @@ message PriceChanged {
 message MarketingNotificationTriggered {
   repeated string user_ids = 1;
   map<string, LocalizedContent> localized_content = 2;
-  optional DeepLink deep_link = 3;
+  optional DeepLink deep_link = 3; // deprecate this? or make a breaking change?
   bool should_send_push = 4;
   bool should_add_to_history = 5;
   bool should_add_to_bulletin = 6;
+  optional Action action = 7;
 }
 
 message LocalizedContent {
   string title = 1;
   string body = 2;
 }
 
+enum ActionType {
+  OPEN_DEEP_LINK = 0;
+  OPEN_EXTERNAL_URL = 1;
+}
+
+message Action {
+  ActionType type = 1;
+  oneof data {
```

Comment:
isn't type redundant when you use oneof?

Reply:
thanks.. fixed it.